### DISCUSSION TITLE
Clean up DNS queries - network.data.decoded

### DISCUSSION THREAD

**Discussion by cac0ns3c** [Security-Onion-Solutions/securityonion#5673] [Unsupported Versions]
Hello everyone,

When i'm seeing a DNS based alert the field network.data.decoded is filled with ....<DOMAIN>....
Example: `.............dev.to.......)........`
But in /nsm/zeek/logs/current/dns.log
cat /nsm/zeek/logs/current/dns.log | grep dev.to
```
{"ts":"2021-09-27T18:24:51.044410Z","uid":"CYu7Ao1NzZnWKUz3wk","id.orig_h":"****","id.orig_p":52369,"id.resp_h":"8.8.4.4","id.resp_p":53,"proto":"udp","trans_id":7913,"query":"dev.to","qclass":1,"qclass_name":"C_INTERNET","qtype":65,"qtype_name":"query-65","rcode":0,"rcode_name":"NOERROR","AA":false,"TC":false,"RD":true,"RA":false,"Z":0,"rejected":false}
{"ts":"2021-09-27T18:24:51.044311Z","uid":"CumqRf2auNhnNLikHd","id.orig_h":"****","id.orig_p":52498,"id.resp_h":"8.8.4.4","id.resp_p":53,"proto":"udp","trans_id":53869,"rtt":0.07411789894104004,"query":"dev.to","qclass":1,"qclass_name":"C_INTERNET","qtype":1,"qtype_name":"A","rcode":0,"rcode_name":"NOERROR","AA":false,"TC":false,"RD":true,"RA":true,"Z":0,"answers":["151.101.130.217","151.101.66.217","151.101.194.217","151.101.2.217"],"TTLs":[300.0,300.0,300.0,300.0],"rejected":false}
```
Wonder what i can do to get the clean domain?

**Comment by clairmont32** [comment]
The alert may contain the extra `.` for the ACII representation of the data on the wire. You should be able to select `hunt` after single left clicking on the source port (for ease of searching the logs) to get all of the network traffic for that source.port value (you may need to adjust the timeframe to match the original alert time) and see the rest of the DNS traffic for that alert. You can play with the Hunt `group by` and search fields if they dont match your desires and save them (docs should provide this data).

You can also find this manually by searching the Kibana > DNS logs with something like `query: dev.to` or `query.keyword: dev.to` (please verify my syntax as I'm typing this from memory atm).

**Comment by dougburks** [comment]
Have you tried using the `Correlate` action on the context menu? It may be able to correlate from the alert to the dns log for you.
https://docs.securityonion.net/en/2.3/alerts.html#actions
